### DISCUSSION TITLE
Correcting version

### DISCUSSION THREAD

**Discussion by yschimke** [renovatebot/renovate#15202] [Suggest an Idea]
If renovate suggests a PR update, is there an easy way to ask it to select a different version.

Currently in the Android Compose world, Kotlin 1.6.10 was the current version.  But 1.6.20 is now the new version.  However Kotlin 1.6.21 is available so renovate opts for that instead.  Can you redirect an inflight PR?

**Comment by rarkins** [comment]
No, that is not a supported feature. Is 1.6.21 considered an officially unstable release or do you just prefer not to use it?

**Reply by yschimke** [reply]
It's this issue, about to hit every compose developer using alphas.  A simple timing incompatibility that developers need to account for.

https://twitter.com/JakeWharton/status/1516836079713857542

**Reply by rarkins** [reply]
You can use `allowedVersions` in your config to restrict allowed versions to update to. Any existing PR will be updated or closed after that. If you think *every* user should avoid 1.6.21 then you could contribute it to our workarounds preset: https://github.com/renovatebot/renovate/blob/main/lib/config/presets/internal/workarounds.ts

But if it's only a subset of users affected then we wouldn't want to apply it by default.

**Reply by yschimke** [reply]
allowedVersions seems right, thanks.

**Comment by yschimke** [comment]
I think the bit I dropped between my original question and the allowedVersions option, was being able to redirect the PR without manual config.

Closing the PR is a good example of this, parks it and stops it re-prompting.  Same with the checkboxes in the github issue descriptions.

But that doesn't work when there is a different version you'd like to update to.  Be nice if there was a conversational way to ask for the PR to target a different version.

**Reply by rarkins** [reply]
I agree, although it's challenging to do within the restrictions of the Pull Request interface. At times the list could be hundreds of releases, for example.

**Reply by yschimke** [reply]
Yeah, I find the docs daunting, because the renovate platform is so rich and evolved.

The more things that can be made user actionable the better.  Also I think some of these are probably shared configs for areas like Android development we just haven't started using.

Should there be an android option in https://github.com/renovatebot/renovate/tree/main/lib/config/presets?

**Reply by rarkins** [reply]
An android option could be great, if you think there are multiple rules/workarounds needed specifically for Android users and most would like to "subscribe" to such a preset

**Reply by rarkins** [reply]
BTW I started https://github.com/renovatebot/renovate/discussions/15221 on the topic of commands

**Reply by yschimke** [reply]
I might start with just clustering typical updates, see what else we should add.  Should I submit a PR?

**Reply by rarkins** [reply]
Maybe an issue or discussion first to clarify

**Reply by yschimke** [reply]
https://github.com/renovatebot/renovate/discussions/15285
